### DISCUSSION TITLE
onedrive下载限速问题的解决方法

### DISCUSSION THREAD

**Discussion by youtehub** [AlistGo/alist#3252] [General]
直接使用rclone挂载的话，使用来实现全速下载，在ubuntu上使用rclone挂载 onedrive测试，onedrive使用的是国际版，生成的 JSON 带有 多个参数，与是否开启 **科学上网** 无关

rclone 配置，使用 rclone config 生成那个完成的token json 是直接使用  [https://tool.nn.ci/onedrive/request](https://tool.nn.ci/onedrive/request) 生成后，按 F12 拿的连接返回的JSON做替换，拿到说现在的alist配置不完整吗？
反正，在rclone生成的配置中，包含这些参数   **drive_id，drive_type ，access_token，refresh_token，"token_type":"Bearer"**  速度就拉满了

有能力的人可以结合 rclone，开发版的 alist 测试一下吧，感觉现在 onedrive 挂载后没有下载速度就是因为参数没有给全

这里是获取 JSON的提取处
![image](https://user-images.githubusercontent.com/18629921/216754250-3f556e7d-3323-408f-a384-08dc2819ef17.png)

这是rclone生成的配置
![image](https://user-images.githubusercontent.com/18629921/216755094-36d7ee88-06a4-405f-8c44-be01c6154604.png)

这是rclone生成的token的完整 JSON 配置项
![image](https://user-images.githubusercontent.com/18629921/216754949-8ebdc280-2d75-4a14-8a98-1b137dc08501.png)

这是使用 FinalShell 的下载速度
![image](https://user-images.githubusercontent.com/18629921/216755130-a1b9fc70-e829-434d-8f09-0e43dd4b7b90.png)

**Comment by youtehub** [comment]
方法二，使用项目组开发的 alist-proxy，在Linux部署，启动命令   ./alist-proxy   -address http://127.0.0.1:5244  -port 35231   -token   令牌值
这种方法也可以直接加速下载

下图中的令牌值
![image](https://user-images.githubusercontent.com/18629921/216764584-83728c60-2269-424b-87b9-7be09991bc27.png)

**Reply by lm317379829** [reply]
本机部署，本机代理？是这个意思吗？这样加速是什么原理？

**Comment by AbyssSkb** [comment]
我的rclone挂载了E5的onedrive根本跑不到满速，用自建API也没用，只有200k。。。

**Reply by youtehub** [reply]
onedrive 就是服务器在外网，我前面尝试后发现，还是得使用 科学上网才行  在ubuntu上使用 shellclash，window 使用 clash 之类的，加了那个加速服务软件 alist proxy  也一样，或者使用官方推荐的 cloudflare
